### ISSUE TITLE
Some enhancements for all UI type

### ISSUE THREAD

**Issue by berkansasmaz** [abpframework/abp#11837]
Related: #11742

**PS, Please see [these notes](https://github.com/abpframework/abp/issues/11837#issuecomment-1081523347) before starting the issue.**

--- 
- [x] Angular (#11840)
- [x] MVC (#11839)
- [x] Blazor ( #11838)
---
1.  The user should be informed that the `Username` cannot contain spaces. Like special char warning in the `Password` input field.

![afbef28a-0518-4e6e-88e9-3bc974b64885](https://user-images.githubusercontent.com/99716457/156318705-6cc9bd47-1838-4d53-954e-34ef7dbcf025.gif)

2. I cannot delete my own user in user management. Therefore, delete should not appear in the action options.

![156362065-f0de8397-ed53-4457-9751-a5fb491af09d (1)](https://user-images.githubusercontent.com/31216880/157043805-06b7ad0a-4eb4-431e-8369-ac68fe7cd3a5.png)

![156370277-740aa335-13b8-4149-9c40-35de993fbae7](https://user-images.githubusercontent.com/31216880/157182777-50070d7d-fd48-498f-9703-18788d7bc189.png)

3. Required fields can be specified with "*" on the settings page.

![156367225-a8f3b80d-0464-4d67-a11a-abacea4bf69f](https://user-images.githubusercontent.com/31216880/157043902-0733de80-5c90-49f3-a22b-106405f4c506.png)

4. A warning message should be given when the user removes all permissions for their role or a warning message may be given when removing all admin user roles. Because it will not be able to make changes to the relevant pages again.

5. The emailing area could be wider, it looks like a button. Maybe if there is only one setting we shouldn't display it as a tab

Also, [see](https://github.com/abpframework/abp/issues/11742#issuecomment-1058364142): 

> The width of the text will depend on the language. For example, in Spanish it may be:
> Envío de correo electrónico
> ![156397387-a0cc8a2a-ec8b-44ac-a6b7-26501945fb90](https://user-images.githubusercontent.com/31216880/157058685-c82f6675-1c5d-4101-a327-164d55a8998c.png)

6. The appearance of the homepage in module templates can be like in app templates.
7. The Hide/Show feature can be added to the entered password wherever the Password input field is located. Related QA Question: https://support.abp.io/QA/Questions/2483

**Comment by EngincanV** [comment]
We've made a meeting and get the below notes:

> 1. The user should be informed that the Username cannot contain spaces. Like special char warning in the Password input field.

* This constraint is coming from the Microsoft Identity library, so we've decided to leave it as it is.

> 2. I cannot delete my own user in user management. Therefore, delete should not appear in the action options.

* Don't show the **Delete** action for the current user on the **Users** page and for the static roles on the **Roles** page.

> 5. The emailing area could be wider, it looks like a button. Maybe if there is only one setting we shouldn't display it as a tab

* We can make the tab name section wider, it's already done in MVC side, we can check for the other UI types.

> 6. The appearance of the homepage in module templates can be like in app templates.

* We can create a new issue for this and assign it to the next milestone. (https://github.com/volosoft/volo/issues/11923)

> 7. The Hide/Show feature can be added to the entered password wherever the Password input field is located. Related QA Question: https://support.abp.io/QA/Questions/2483

* We might need to take guidance from @armgnunlu for this issue, because **Microsoft Edge** shows a hide/show password icon by default for the password input type.

**Comment by hikalkan** [comment]
I think we can close this. We can follow with https://github.com/abpframework/abp/issues/11840
